### PR TITLE
[PW-8205] Abstract JS PM handler and implement pm specific JS files. 

### PR DESCRIPTION
**Pull request by raoulritter**
**Description**

Moved from same configuration to one parent configuration for generic payment method and extending this when necessary. 

**Tested scenarios**
- Each payment method.

### CODE REVIEW

File: view/frontend/web/js/view/payment/method-renderer/adyen-googlepay-method.js [hunk 1]
```diff
@@ -21,70 +21,9 @@ define(
...
-                        street: formattedShippingAddress.street
-                    };
-                }
-
-                // GooglePay requires extra configuration
-                //configuration = Object.assign(configuration, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
-
-                return configuration;
+                let baseComponentConfiguration = this._super();
+                let googlePayConfiguration = Object.assign(baseComponentConfiguration.showPayButton=true, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration,);
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-googlepay-method.js:25]:
```suggestion
                let googlePayConfiguration = Object.assign(baseComponentConfiguration.showPayButton=true, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js [hunk 2]
```diff
@@ -21,66 +21,9 @@ define(
...
-                        street: formattedShippingAddress.street
-                    };
-                }
-
-                // Paypal requires extra configuration
-                configuration = Object.assign(configuration, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
-
-                return configuration;
+                let baseComponentConfiguration = this._super();
+                let paypalConfiguration = Object.assign(baseComponentConfiguration.showPayButton=true, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration,);
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:25]:
```suggestion
                let paypalConfiguration = Object.assign(baseComponentConfiguration.showPayButton=true, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js [hunk 3]
```diff
@@ -554,6 +533,47 @@ define(
...
+                    configuration.data.shippingAddress = {
+                        city: formattedShippingAddress.city,
+                        country: formattedShippingAddress.country,
+                        houseNumberOrName: formattedShippingAddress.houseNumber,
+                        postalCode: formattedShippingAddress.postalCode,
+                        street: formattedShippingAddress.street
+                    };
+                }
+                return configuration
+
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js:575]:
```suggestion
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-googlepay-method.js [hunk 4]
```diff
@@ -21,70 +21,9 @@ define(
...
-                        street: formattedShippingAddress.street
-                    };
-                }
-
-                // GooglePay requires extra configuration
-                //configuration = Object.assign(configuration, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
-
-                return configuration;
+                let baseComponentConfiguration = this._super();
+                let googlePayConfiguration = Object.assign(baseComponentConfiguration.showPayButton=true, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
```
Comment (peterojo) [inline view/frontend/web/js/view/payment/method-renderer/adyen-googlepay-method.js:25]: Could you move the `showPayButton` assignment to a different line?
```suggestion
                baseComponentConfiguration.showPayButton = true;
                let googlePayConfiguration = Object.assign(baseComponentConfiguration, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js [hunk 5]
```diff
@@ -21,66 +21,9 @@ define(
...
-                        street: formattedShippingAddress.street
-                    };
-                }
-
-                // Paypal requires extra configuration
-                configuration = Object.assign(configuration, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
-
-                return configuration;
+                let baseComponentConfiguration = this._super();
+                let paypalConfiguration = Object.assign(baseComponentConfiguration.showPayButton=true, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
```
Comment (peterojo) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:25]: Could you move the `showPayButton` assignment to a different line?
```suggestion
                baseComponentConfiguration.showPayButton = true;
                let googlePayConfiguration = Object.assign(baseComponentConfiguration, paymentMethodsExtraInfo[paymentMethod.methodIdentifier].configuration);
```

File: view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js [hunk 6]
```diff
@@ -554,6 +533,36 @@ define(
                     telephone: address.telephone
                 };
             },
+            buildComponentConfiguration: function (paymentMethod, paymentMethodsExtraInfo, result) {
+                var self = this;
+                var email = '';
+                var showPayButton = false;
```
Comment (peterojo) [inline view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js:539]: this line can be removed and we can hardcode `false` in the configuration object below

File: view/frontend/web/js/view/payment/method-renderer/adyen-googlepay-method.js (hunk 4 above)
Comment (peterojo) [reply view/frontend/web/js/view/payment/method-renderer/adyen-googlepay-method.js:25]: discussed and you can ignore this if you want

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js (hunk 5 above)
Comment (peterojo) [reply view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:25]: discussed and you can ignore this if you want

**Review by peterojo — DISMISSED** [review]

File: view/frontend/web/js/view/payment/method-renderer/adyen-amazonpay-method.js [hunk 7]
```diff
@@ -21,110 +21,75 @@ define(
...
-                var email = '';
-                var showPayButton = true;
-
-                if (!!quote.guestEmail) {
-                    email = quote.guestEmail;
-                }
-                var formattedShippingAddress = {};
-                var formattedBillingAddress = {};
-
+                const amazonSessionKey = 'amazonCheckoutSessionId'
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-amazonpay-method.js:24]: This variable was declared but not used.

File: view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js (hunk 6 above)
Comment (raoulritter) [reply view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js:539]: I think we should keep this for visibility for when we have to add new methods.

File: view/frontend/web/js/view/payment/method-renderer/adyen-amazonpay-method.js (hunk 7 above)
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-amazonpay-method.js:24]:
```suggestion
```

**Review by candemiralp — APPROVED** [review]

**Pull request merged**